spec/models/metric/capture_spec.rb

```diff
@@ -5,6 +5,26 @@
     @zone = EvmSpecHelper.local_miq_server.zone
   end
 
+  describe ".perf_capture_timer" do
+    let(:ems) { FactoryBot.create(:ext_management_system) }
+
+    before { allow(ExtManagementSystem).to receive(:find).with(ems.id).and_return(ems) }
```

Comment:
can this work without this `before`?

Reply:
No because the method is doing `ExtManagementSystem.find(ems_id)` which returns a new object not the one we have in `let` so the `expect()` doesn't match

Reply:
From my experience it seems like the answer to this issue is either this or `allow_any_instance_of(ExtManagementSystem)`

Reply:
yea. I kinda like `allow_any_instance` but rubocop freaks

Reply:
or the other option is move the EMS.find into a method and then use a stub on that method 🤷  - nothing is "nice"